main.py

```diff
@@ -0,0 +1,97 @@
+from fastapi import FastAPI, Depends
+from sqlalchemy.orm import Session
+
+import crud
+import schemas
+from database import SessionLocal
+
+app = FastAPI()
+
+
+def get_db() -> Session:
+    db = SessionLocal()
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@app.post("/authors/", response_model=schemas.Authors)
+def create_author(
+        author: schemas.AuthorCreate,
+        db: Session = Depends(get_db)
+):
+    return crud.create_author(db=db, author=author)
+
+
+@app.get("/authors/", response_model=list[schemas.Authors])
+def list_authors(db: Session = Depends(get_db)):
+    return crud.get_author_list(db=db)
+
+
+@app.get("/authors/{author_id}/", response_model=schemas.Authors)
+def detail_author(
+        author_id: int,
+        db: Session = Depends(get_db)
+):
+    return crud.get_author_by_id(db=db, author_id=author_id)
+
+
+@app.put("/authors/{author_id}/", response_model=schemas.Authors)
+def update_author(
+        author_id: int,
+        author: schemas.AuthorUpdate,
+        db: Session = Depends(get_db)
+):
+    return crud.update_author(db=db, author_id=author_id, author=author)
+
+
+@app.delete("/authors/{author_id}/", response_model=schemas.DeleteResponse)
+def delete_author(
+        author_id: int,
+        db: Session = Depends(get_db)
+):
+    crud.delete_author(db=db, author_id=author_id)
+    return {"message": "Author successfully deleted", "deleted_id": author_id}
+
+
+@app.post("/books/", response_model=schemas.Books)
```

Comment:
The response model for the `create_book` endpoint should be `schemas.Book` instead of `schemas.Books` to match the Pydantic model naming convention.